packages/editor/src/components/block-list/block.js

```diff
@@ -403,11 +403,10 @@ export class BlockListBlock extends Component {
 		const shouldShowMobileToolbar = shouldAppearSelected;
 		const { error, dragging } = this.state;
 
-		// Insertion point can only be made visible when the side inserter is
-		// not present, and either the block is at the extent of a selection or
-		// is the first block in the top-level list rendering.
-		const shouldShowInsertionPoint = ( isPartOfMultiSelection && isFirst ) || ! isPartOfMultiSelection;
```

Comment:
I think this was wrong before; `isFirst` means first of an entire `rootClientId` context, not first in a multi-selection.

https://github.com/WordPress/gutenberg/blob/b022ad2ac66d47c3bbc22f1933d4c11dd465805b/packages/editor/src/components/block-list/index.js#L218